x/ibc/core/03-connection/keeper/handshake.go

```diff
@@ -89,10 +91,12 @@ func (k Keeper) ConnOpenTry(
 		// counterparty is chainA and connection is on INIT stage.
 		// Check that existing connection versions for initialized connection is equal to compatible
 		// versions for this chain.
+		// ensure that existing connection's delay period is the same as desired delay period.
 		if !(previousConnection.Counterparty.ConnectionId == "" &&
 			bytes.Equal(previousConnection.Counterparty.Prefix.Bytes(), counterparty.Prefix.Bytes()) &&
 			previousConnection.ClientId == clientID &&
-			previousConnection.Counterparty.ClientId == counterparty.ClientId) {
+			previousConnection.Counterparty.ClientId == counterparty.ClientId &&
+			previousConnection.DelayPeriod == delayPeriod) {
```

Comment:
NOTE: In case of crossing hellos, both initiators must choose same delay period value